pkg-manager/core/test/catalogs.ts

```diff
@@ -1193,6 +1193,58 @@ describe('add', () => {
     })
   })
 
+  // Regression test for https://github.com/pnpm/pnpm/issues/9759
+  test('adding new usage of default catalog does not mutate catalog entries', async () => {
```

Comment:
I made sure this test fails before the changes in this PR.

<img width="842" height="474" alt="Image" src="https://github.com/user-attachments/assets/fc953fde-a6d2-4e85-8ef1-11a4678112d5" />